src/main/JayZ.ts

```diff
@@ -1,49 +1,123 @@
-import { EncryptedJayZItem, Encryptor } from "./encryptors"
+import { memzero, ready } from "libsodium-wrappers"
+import { DataKey, DataKeyProvider } from "./DataKeyProvider"
+import { Encryptor } from "./Encryptor"
+import { LibsodiumEncryptor } from "./LibsodiumEncryptor"
+import { EncryptedItemMetadataV1, EncryptedJayZItem, LegacyEncryptedItemMetadata, MetadataVersion } from "./types"
 
-export interface JayZProps {
-  encryptor: Encryptor
+export type JayZConfig = {
```

Comment:
Not only that but the type name changes as well from `JayzProps` to `JayZConfig` which is also "breaking". And yea since this package is sub 1.0, breaking changes are 'allowed'. 

After this is merged you might consider just ticking the version to 1.0.

Reply:
I'm confused about this being a breaking change if the currently published version has `JayZConfig` with a required `keyProvider` param. wouldn't we want to preserve that over the type definition that exists on the master branch?

Reply:
@kedskeds I think what you currently have in this PR is what we want: `JayZConfig` with a required `keyProvider`.